### PR TITLE
Password validation

### LINKED ISSUE #3: Add password validation to auth forms
Right now we don't actually have password strength validation. Supabase requires a password of 6 characters so we need to add that validation at the very least.

Right now if Supabase throws an error because the password isn't long enough, the user gets no information.

### PR DESCRIPTION
**Pull request by idiglove**
- [x] Resolves #3 

image

### CODE REVIEW

File: src/components/SignUpForm.tsx [hunk 1]
```diff
@@ -15,7 +15,12 @@ const SignUp: Component = () => {
       initialValues={{ email: '', password: '' }}
       validation={{
         email: Yup.string().email().required(),
-        password: Yup.string().required(),
+        password: Yup.string()
+          .required('Please enter your password')
+          .matches(
+            /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{6,})/,
```
Comment (amorriscode) [inline src/components/SignUpForm.tsx]: This is some magical looking RegEx! 🪄

**Review by amorriscode — COMMENTED** [review]
Thanks for the help @idiglove! 🙏🏻 

How would you feel about trying [`yup-password`](https://github.com/knicola/yup-password)? I just noticed it. The reason I like it is there are multiple, separate messages. With the current solution, the user might not know which password requirement they actually violated.

File: src/components/SignUpForm.tsx (hunk 1 above)
Comment (idiglove) [reply src/components/SignUpForm.tsx]: Ya i saw yup-password as well, but wasnt sure if you guys wanted another package. I can try that out! :)

Comment (amorriscode) [reply src/components/SignUpForm.tsx]: Sure, I'm all for using packages. In an ideal world, I'm a 10x developer that writes everything myself. In this one-product-a-mont-world, let's build on the shoulders of giants! 😅

**Review by amorriscode — APPROVED** [review]
Thanks so much for this @idiglove!

**Pull request merged**